control/control.go

```diff
@@ -603,14 +696,36 @@ func (p *pluginControl) ExpandWildcards(ns core.Namespace) ([]core.Namespace, se
 	return nss, nil
 }
 
-func (p *pluginControl) ValidateDeps(mts []core.Metric, plugins []core.SubscribedPlugin) []serror.SnapError {
+func (p *pluginControl) ValidateDeps(requested []core.RequestedMetric, plugins []core.SubscribedPlugin, configTree *cdata.ConfigDataTree) []serror.SnapError {
 	var serrs []serror.SnapError
-	for _, mt := range mts {
-		errs := p.validateMetricTypeSubscription(mt, mt.Config())
-		if len(errs) > 0 {
-			serrs = append(serrs, errs...)
+	var mts []core.Metric
+	for _, m := range requested {
+		nss, err := p.MatchQueryToNamespaces(m.Namespace())
+		if err != nil {
+			// use metric directly from the workflow
+			nss = []core.Namespace{m.Namespace()}
+		}
+
+		for _, ns := range nss {
+			mts = append(mts, &metric{
+				namespace: ns,
+				version:   m.Version(),
+				config:    configTree.Get(ns.Strings()),
+			})
 		}
+
 	}
+	// to prevent errors for control in distributed workflow
+	// where only processors/publishers loaded (empty metric catalog)
+	if len(p.metricCatalog.Keys()) > 0 {
+		for _, mt := range mts {
+			errs := p.validateMetricTypeSubscription(mt, mt.Config())
+			if len(errs) > 0 {
+				serrs = append(serrs, errs...)
+			}
+		}
+	}
+
```

Comment:
What do you think about move this checkingof the content of metricCatalog (Line 720) into `for` range over `requested` metrics? What is more, I suggest change it to simply checking `if metricCatalog is empty`, and if yes, return an error - because that means that requested metrics cannot be found there.